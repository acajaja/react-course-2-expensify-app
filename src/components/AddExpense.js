import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.addExpense(expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <section>
                    <h1>Add New Expense</h1>
                    <ExpenseForm
                        onSubmit={this.onSubmit} />
                </section>
            </div>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpense);
