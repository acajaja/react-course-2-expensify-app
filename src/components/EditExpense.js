import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpense extends React.Component {
    onClick = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    onSubmit = (updates) => {
        this.props.editExpense(this.props.expense.id, updates);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onClick} title="Remove this expense">x</button>
            </div>
        );
    };
};

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    editExpense: (id, updates) => dispatch(editExpense(id, updates)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);
