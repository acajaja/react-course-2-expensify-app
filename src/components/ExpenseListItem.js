import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <section>
            <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
            <p>
                {numeral(amount / 100).format('$0,0.00')}
                -
                created at: {moment(createdAt).format('MMMM Do, YYYY')}
            </p>
        </section>
    </div>
);

export default ExpenseListItem;
