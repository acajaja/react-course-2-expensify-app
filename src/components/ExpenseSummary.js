import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({expensesCount, totalAmount}) => {
    return (
        <div>
            <p>
                There are {expensesCount} expenses
                totaling {numeral(totalAmount / 100).format('$0,0.00')}.
            </p>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: visibleExpenses.length,
        totalAmount: expensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
