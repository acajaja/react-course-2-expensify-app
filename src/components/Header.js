import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = (props) => (
    <header>
        <h1>Expensify</h1>
        <nav>
            <NavLink to="/" activeClassName="is-active" title="Home" exact={true}>Home</NavLink>
            <NavLink to="/create" activeClassName="is-active" title="Create">Create</NavLink>
            <NavLink to="/edit" activeClassName="is-active" title="Edit">Edit</NavLink>
            <NavLink to="/help" activeClassName="is-active" title="Help">Help</NavLink>
        </nav>
    </header>
);

export default Header;
