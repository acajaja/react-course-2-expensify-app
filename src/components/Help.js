import React from 'react';

const HelpPage = () => (
    <div>
        <p>HelpPage component!</p>
    </div>
);

export default HelpPage;
