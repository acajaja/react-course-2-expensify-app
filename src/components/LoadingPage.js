import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" alt="Loading..." src="/images/loader.gif" />
    </div>
);

export default LoadingPage;
