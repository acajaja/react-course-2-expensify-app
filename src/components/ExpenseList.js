import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="content-container">
        <section>
            <div className="list-header">
                <div className="show-for-mobile">Expenses</div>
                <div className="show-for-desktop">Expense</div>
                <div className="show-for-desktop">Amount</div>
            </div>
            <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item--message">
                        <p>No expenses</p>
                    </div>
                ) : (
                    props.expenses.map((expense) => (
                        <ExpenseListItem key={expense.id} {...expense} />
                    ))
                )
            }
            </div>
        </section>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
