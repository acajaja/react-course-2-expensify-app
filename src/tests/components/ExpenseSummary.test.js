import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('Test a single expense', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={1} totalAmount={123} />);
    expect(wrapper).toMatchSnapshot();
});

test('Test multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expensesCount={16} totalAmount={1345223} />);
    expect(wrapper).toMatchSnapshot();
});
