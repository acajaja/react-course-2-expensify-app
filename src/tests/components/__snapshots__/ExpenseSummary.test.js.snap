// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test a single expense 1`] = `
<div>
  <p>
    There are 
    1
     expenses totaling 
    $1.23
    .
  </p>
</div>
`;

exports[`Test multiple expenses 1`] = `
<div>
  <p>
    There are 
    16
     expenses totaling 
    $13,452.23
    .
  </p>
</div>
`;
